pkg/grok/docker_file.go

```diff
@@ -10,41 +10,66 @@ import (
 )
 
 // ResolveDockerfileDependencies given a docker file, resolve its dependencies
-func ResolveDockerfileDependencies(path string) (string, []string, error) {
+func ResolveDockerfileDependencies(buildArgs []string, path string) (string, []string, error) {
 	file, err := os.Open(path)
 	if err != nil {
 		return "", nil, fmt.Errorf("Error opening dockerfile %s, error: %s", path, err)
 	}
 	scanner := bufio.NewScanner(file)
-
+	context, err := parseBuildArgs(buildArgs)
+	if err != nil {
+		return "", nil, err
+	}
 	originLookup := map[string]string{} // given an alias, look up its origin
 	allOrigins := map[string]bool{}     // set of all origins
 	var origin string
 	for scanner.Scan() {
 		line := scanner.Text()
 		tokens := strings.Fields(line)
-		if len(tokens) > 0 && strings.ToLower(tokens[0]) == "from" {
-			if len(tokens) < 2 {
-				return "", nil, fmt.Errorf("Unable to understand line %s", line)
-			}
-
-			var image = tokens[1]
-			var found bool
-			origin, found = originLookup[image]
-			if !found {
-				allOrigins[image] = true
-				origin = image
-			}
-
-			if len(tokens) > 2 {
-				if len(tokens) < 4 || strings.ToLower(tokens[2]) != "as" {
+		if len(tokens) > 0 {
+			switch strings.ToUpper(tokens[0]) {
+			case "FROM":
+				if len(tokens) < 2 {
 					return "", nil, fmt.Errorf("Unable to understand line %s", line)
 				}
-				alias := tokens[3]
-				originLookup[alias] = origin
-				// Just ignore the rest of the tokens...
-				if len(tokens) > 4 {
-					logrus.Debugf("Ignoring chunks from FROM clause: %v", tokens[4:])
+				var image = os.Expand(tokens[1], func(key string) string {
```

Comment:
Cool it looks os.Expand will handle $arg, ${arg} very well?

Reply:
yes, I used it in build.Context too. Just forgot about it at the moment during our offline conversation.

As you can see both $arg and ${arg} are used in our test example.